src/net_processing.cpp

```diff
@@ -3969,11 +3969,58 @@ void PeerManager::ConsiderEviction(CNode& pto, int64_t time_in_seconds)
 
 void PeerManager::EvictExtraOutboundPeers(int64_t time_in_seconds)
```

Comment:
`EvictExtraOutboundPeers()` seems like a natural candidate to be separated into two functions,`EvictExtraOutboundBlockRelayPeers()` and `EvictExtraOutboundFullRelayPeers()`. This might enable better fuzzing or unit testing of the logic. If you agree, it may be easier to do this now than later in a refactoring-for-testability PR.

Reply:
Well, there's also an argument to be made that it's simpler to have all of the outbound peer eviction logic in one place/one callback.

If there is a reason (like testing) to do so, I think splitting this up in the future could make sense, but for now I'd like to keep the changes here to be more straightforward.  So assuming this implementation is correct I'd prefer to leave it as-is.

Reply:
Feel free to pick this up in a future PR if you like; I'm leaving this as-is for now as I think this is easier to review and understand if this all happens in the same scheduler callback.